### PR TITLE
Added touchmove support to slider component

### PR DESCRIPTION
**Pull request by keul**
See #5855

**Note**: the pull request is for the master branch, but I'm wondering if another one can be accepted for the old 1.x branch

### CODE REVIEW

**Comment by alexcjohnson** [comment]
@keul this looks good! Re: 1.x - we'd really rather keep our efforts focused on v2 except for security fixes. Is there something specific preventing you from updating to v2?

The one thing we should add is a test. Perhaps we could copy or extend this test:

https://github.com/plotly/plotly.js/blob/e27ca3c691924b760a697f7176854cd91ae4dcd5/test/jasmine/tests/sliders_test.js#L470-L508

Note we have `assets/touchEvent` (and `assets/mouseEvent`) which were probably created after `sliders_test` but help to ensure these events are more realistic without making them harder to construct, like here in the shape drawing test:

https://github.com/plotly/plotly.js/blob/e27ca3c691924b760a697f7176854cd91ae4dcd5/test/jasmine/tests/draw_newshape_test.js#L11-L38

**Comment by keul** [comment]
@alexcjohnson
> Re: 1.x - we'd really rather keep our efforts focused on v2 except for security fixes. Is there something specific preventing you from updating to v2?

Nothing but not knowing the compatibility with serve side modules. My use of plotly.js depends on the plotly Python module so I try to keep the same version distributed with the lib itself.
But this is not a problem: I can continue providing my patch until I finally migrate to v2.

----

I'll take a look adding tests (not sure I'll be able to solve this before summer vacation). Thanks for references!

**Comment by archmoj** [comment]
Thanks for the PR.

Please create `draftlogs/5856_add.md` as described in this [README](https://github.com/plotly/plotly.js/blob/master/draftlogs/README.md).
Something like: 
```
 - Add touch support to `slider` component [[#5856](https://github.com/plotly/plotly.js/pull/5856)], 
   with thanks to @keul for the contribution!
```

File: test/jasmine/assets/touch_event.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ module.exports = function(type, x, y, opts) {
         target: el,
         clientX: x,
         clientY: y,
+        screenX: x,
```
Comment (keul) [inline test/jasmine/assets/touch_event.js:12]: I was forced to complete the signature of the event or simulating the touch event on the slider will generate a very strange behavior (slider was always moved on the start position).

**Comment by keul** [comment]
@alexcjohnson I added the test.

But now there's another test that is failing: https://github.com/plotly/plotly.js/pull/5856/commits/5fe96ec988c9b8f69b992b9dffed9c6c6021d88a#diff-09276d81835ab1fc83e8eb085e4faae2de13348f4be40367cb03276a3f651cbfR549

Seems like `should respond to touch interactions` is not really an isolated test. It starts to fails as soon as I add this new test, and it's back to normal when removing it, but I'm not able to understand what's wrong.

**Comment by archmoj** [comment]
> @alexcjohnson I added the test.
> 
> But now there's another test that is failing: [5fe96ec#diff-09276d81835ab1fc83e8eb085e4faae2de13348f4be40367cb03276a3f651cbfR549](https://github.com/plotly/plotly.js/commit/5fe96ec988c9b8f69b992b9dffed9c6c6021d88a#diff-09276d81835ab1fc83e8eb085e4faae2de13348f4be40367cb03276a3f651cbfR549)
> 
> Seems like `should respond to touch interactions` is not really an isolated test. It starts to fails as soon as I add this new test, and it's back to normal when removing it, but I'm not able to understand what's wrong.

What happens if you add it to the end?

**Comment by keul** [comment]
> What happens if you add it to the end?

@archmoj still fails

**Comment by archmoj** [comment]
The test passes on my machine.
To get all the greens, please fetch `upstream/master` in your fork, then pull and merge it into this branch.

**Comment by keul** [comment]
@archmoj really sorry for the noise… It works now 🙇

File: test/jasmine/assets/touch_event.js (hunk 1 above)
Comment (archmoj) [reply test/jasmine/assets/touch_event.js:12]: Do we really need to add these? It looks the test passes without it.

Comment (keul) [reply test/jasmine/assets/touch_event.js:12]: @archmoj not on my machine (MacOS). Just tested again removing these lines and I have the failure i fixed adding this

Comment (keul) [reply test/jasmine/assets/touch_event.js:12]: BTW: I'm able to reproduce the issue also outside Jasmine/jsdom.

issue

I'm 100% for removing this if you think it's just an issue on my Chrome… but seems really strange to me.

Comment (archmoj) [reply test/jasmine/assets/touch_event.js:12]: Thanks for testing.
LGTM :medal_sports: 
Over to @alexcjohnson

**Review by alexcjohnson — APPROVED** [review]
Love it. 💃

**Comment by archmoj** [comment]
We will wait for @nicolaskruchten before merging.

**Pull request merged**